portal-web/docroot/html/taglib/theme/portlet_messages/page.jsp

```diff
@@ -97,6 +97,6 @@ else if (group.isStagingGroup()) {
 <liferay-ui:success key="<%= portlet.getPortletId() + SessionMessages.KEY_SUFFIX_UPDATED_CONFIGURATION %>" message="you-have-successfully-updated-the-setup" />
 <liferay-ui:success key="<%= portlet.getPortletId() + SessionMessages.KEY_SUFFIX_UPDATED_PREFERENCES %>" message="you-have-successfully-updated-your-preferences" />
 
-<c:if test="<%= !MultiSessionErrors.contains(renderRequest, NoSuchLayoutException.class.getName()) && !MultiSessionMessages.contains(renderRequest, portlet.getPortletId() + SessionMessages.KEY_SUFFIX_HIDE_DEFAULT_ERROR_MESSAGE) %>">
+<c:if test="<%= Objects.equals(portlet.getPortletName(), PortalUtil.getPortletId(request)) && !MultiSessionErrors.contains(renderRequest, NoSuchLayoutException.class.getName()) && !MultiSessionMessages.contains(renderRequest, portlet.getPortletId() + SessionMessages.KEY_SUFFIX_HIDE_DEFAULT_ERROR_MESSAGE) %>">
```

Comment:
@holatuwol I agree with you in https://github.com/liferay-frontend/liferay-portal/pull/1104#discussion_r648065582, this looks risky and with possible uncovered edge cases.

Maybe the way to go is to treat password exception the same way `NoSuchLayoutException` is handled here, something like 
```jsp
<c:if test="<%= !MultiSessionErrors.contains(renderRequest, UserPasswordException.MustNotBeEqualToCurrent.class.getName()) && ...
```
I tested this and it prevents all toast messages, leaving only the inline one. This to me seems like the correct behavior, rather then less worse showing of only one toast.

I am not sure if there is a simple way in Java to cover all subtypes of `UserPasswordException`.

What do you think?

Reply:
It's a little weird since everything is a string, rather than class names. However, since all of the subtypes of UserPasswordException are inside the class itself, we can add all the declared classes to the checks. I've pushed an update which uses that check instead.